pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -170,6 +170,17 @@ type ScorePlugin interface {
 	Score(pc *PluginContext, p *v1.Pod, nodeName string) (int, *Status)
 }
 
+// ScoreWithNormalizePlugin is an interface that must be implemented by "score"
+// plugins that also need to normalize the node scoring results produced by the same
+// plugin's "Score" method.
+type ScoreWithNormalizePlugin interface {
```

Comment:
Since we have tight coupling between score and normalize, I think we should treat this as a special type of Score plugin.

I suggest we remove the explicit NormalizeScore list from the [plugins struct](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/types.go). A ScoreWithNormalizePlugin can be added in the Score PluginSet.

For each entry in the Score PluginSet, the framework initialization code will basically first make sure that it implements the Score interface and error out if it does not. Next it checks if it also implements the ScoreWithNormalizePlugin interface, if it does, then we add it to the internal normalize plugins list, if not, then we just do nothing.

Reply:
This approach is fine with me. Compared to the current approach, this has the limitation of not being able to disable "Normalize Score" plugins independently of Score plugins, but I doubt if users really need to disable "Normalize" only and leave "Score" in place.